web-frontend/src/pages/TransferCoins/SendCoins/SendForm.tsx

```diff
@@ -76,15 +81,24 @@ export function SendForm(props: SendFormProps) {
   useEffect(() => {
     if (!sendOpData) {
       setAmount(redirectAmount);
-      setRecipient(redirectedTo);
       setFees(PRESET_LOW);
+      setRecipient(redirectedTo);
     } else {
-      setAmount(sendOpData?.amount);
-      setRecipient(sendOpData?.recipientAddress);
-      setFees(sendOpData?.fees);
+      if (sendOpData.recipientDomainName) {
+        setRecipient(sendOpData.recipientDomainName + mnsExtension);
+        resolveDns(sendOpData.recipientDomainName);
+      } else {
+        setRecipient(sendOpData.recipientAddress);
+      }
+      setAmount(sendOpData.amount);
+      setFees(sendOpData.fees);
+      setSelectedAsset(sendOpData.asset);
     }
   }, [sendOpData, redirectAmount, redirectedTo]);
```

Comment:
React Hook useEffect has a missing dependency: 'resolveDns'. Either include it or remove the dependency array